associated-token-account/program/src/lib.rs

```diff
@@ -21,13 +21,11 @@ pub(crate) fn get_associated_token_address_and_bump_seed(
     spl_token_mint_address: &Pubkey,
     program_id: &Pubkey,
 ) -> (Pubkey, u8) {
-    Pubkey::find_program_address(
-        &[
-            &wallet_address.to_bytes(),
-            &spl_token::id().to_bytes(),
-            &spl_token_mint_address.to_bytes(),
-        ],
+    get_associated_token_address_and_bump_seed_internal(
```

Comment:
I believe these and following `./program` diffs should absolutely be no-functional-changes. :)

Reply:
Well, associated token program actually gets more flexible because it will not error for non-Tokenkeg program id's. But I think that's a nice change :) Do you mind adding the program change details to the PR title and description for posterity?

Reply:
yeah, that's right. I meant in the context of real production environment (mainnet-beta). :)

Reply:
Also, the subtlety is that deployed program and this helper `get_associated_token_address` are still hard-coded to the Tokenkeg program id...

Reply:
Ah, I thought you planned to deploy an update to associated token program at some point. Are the program changes just for local testing then?

Reply:
@jstarry According to this (https://explorer.solana.com/address/ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL),  the associated token program isn't upgradable? ;) I wonder it's worth it... Also, I came up with some niceties: #1371